engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/BpmnEventTypeTest.java

```diff
@@ -380,6 +380,44 @@ BpmnModelInstance modelInstance() {
                   .endEvent(elementId(), e -> e.escalation(escalationCode()))
                   .done();
             }
+          },
+          new BpmnEventTypeScenario(
+              "Compensation Intermediate Throw Event",
+              BpmnElementType.INTERMEDIATE_THROW_EVENT,
+              BpmnEventType.COMPENSATION) {
+            @Override
+            BpmnModelInstance modelInstance() {
+              return Bpmn.createExecutableProcess(processId())
+                  .startEvent()
+                  .intermediateThrowEvent(
+                      elementId(),
+                      t -> t.compensateEventDefinition().compensateEventDefinitionDone())
+                  .endEvent()
+                  .done();
+            }
+          },
+          new BpmnEventTypeScenario(
+              "Compensation End Event", BpmnElementType.END_EVENT, BpmnEventType.COMPENSATION) {
+            @Override
+            BpmnModelInstance modelInstance() {
+              return Bpmn.createExecutableProcess(processId())
+                  .startEvent()
+                  .endEvent(
+                      elementId(),
+                      end -> end.compensateEventDefinition().compensateEventDefinitionDone())
+                  .done();
+            }
+          },
+          new BpmnEventTypeScenario(
+              "Compensation Start Event", BpmnElementType.START_EVENT, BpmnEventType.COMPENSATION) {
+            @Override
+            BpmnModelInstance modelInstance() {
+              return Bpmn.createExecutableProcess(processId())
+                  .startEvent(elementId())
+                  .compensation()
+                  .endEvent()
+                  .done();
+            }
```

Comment:
:x: Invalid process. A compensation start event should be only allowed for event subprocesses within embedded subprocesses.

Since we can't trigger a compensation event subprocess at the moment, we can ignore this case.

Reply:
I wasn't sure about this test case. Thank you